src/map/clif.cpp

```diff
@@ -5031,6 +5031,16 @@ void clif_getareachar_unit( map_session_data* sd,block_list *bl ){
 		return;
 	}
 
+	// Prevents visual bug where monster shadows appear when entering view range
+	if (battle_config.hide_cloaked_units) {  
```

Comment:
You can just check `battle_config.hide_cloaked_units&bl->type` here directly, then you don't need the if below and save processing time too.